### PR TITLE
5893 add archivedDocuments array to case and case internal

### PR DESCRIPTION
**Pull request by mrinsin**

### CODE REVIEW

File: shared/src/business/entities/cases/CaseInternal.js [hunk 1]
```diff
@@ -28,7 +29,7 @@ function CaseInternal(rawCase, { applicationContext }) {
   if (!applicationContext) {
     throw new TypeError('applicationContext must be defined');
   }
-
+  this.archivedDocuments = rawCase.archivedDocuments;
```
Comment (kkoskelin) [inline shared/src/business/entities/cases/CaseInternal.js]: Do we want this here if we're setting it later (line 81)?

File: shared/src/business/entities/cases/Case.js [hunk 2]
```diff
@@ -238,6 +239,14 @@ function Case(rawCase, { applicationContext, filtered = false }) {
     this.documents = [];
   }
 
+  if (Array.isArray(rawCase.archivedDocuments)) {
+    this.archivedDocuments = rawCase.archivedDocuments
+      .map(document => new Document(document, { applicationContext }))
+      .sort((a, b) => compareStrings(a.createdAt, b.createdAt));
```
Comment (kkoskelin) [inline shared/src/business/entities/cases/Case.js]: Sorting frequently is found later in helpers or computeds. Will this be useful here?

**Review by kkoskelin — CHANGES_REQUESTED** [review]
One item with an assignment that needs tweaking, I think. The rest are just my opinions. :D

File: shared/src/business/entities/cases/CaseInternal.js (hunk 1 above)
Comment (mrinsin) [reply shared/src/business/entities/cases/CaseInternal.js]: good point... I will remove it!

**Review by kkoskelin — APPROVED** [review]

**Pull request merged**